config-service-impl/src/main/java/org/hypertrace/config/service/ConfigServiceGrpcImpl.java

```diff
@@ -106,9 +106,12 @@ public void getAllConfigs(
   public void deleteConfig(
       DeleteConfigRequest request, StreamObserver<DeleteConfigResponse> responseObserver) {
     try {
+      ConfigResource configResource = getConfigResource(request);
```

Comment:
This will error if the value didn't exist, which is different than the previous behavior, right? That seems more correct to me, just want to point it out.

Reply:
No, why will this line throw an error?

I'm assuming you wanted to highlight the next line - #110 ? (i.e. `configStore.getConfig()` would throw an error if the value didn't exist?) - but that's not right - `configStore.getConfig()` would return empty config value in that case (and so, would this `deleteConfig` API).

Reply:
sorry, didn't dive in - I thought getConfig throws if resource isn't present, but maybe that logic comes in at a higher layer. I'm not sure which is preferable, but it seems wrong to write an empty if no value exists, no?

Reply:
> I'm not sure which is preferable, but it seems wrong to write an empty if no value exists, no?

We're treating empty config value as equivalent to no config value (in the store). So, going by that, there is no difference if no value exists or empty value exists in the store.

Reply:
but wouldn't this do things like increment the version and set a create time, so when we do write a config for the first time, its metadata will be off?

Reply:
Actually, I take that back. Thinking from the API perspective - deleting a non-existing(whether no config value was defined or the config value was deleted previously) config should throw a `Status.NOT_FOUND` exception, right? So, if `configStore.getConfig()`(line #110) returns an empty config, then we should throw that exception, right?

Reply:
Sorry, I just saw your comment after I typed my above comment. 

> but wouldn't this do things like increment the version and set a create time, so when we do write a config for the first time, its metadata will be off?

Agree. I'll make the above change then.

Reply:
> deleting a non-existing(whether no config value was defined or the config value was deleted previously) config should throw a Status.NOT_FOUND exception, right?

Yeah that was the other part I was kind of toying with above, but that behavior seems right to me.

Reply:
Made this change. Also added a unit test for deleting a non-existing config.